windows/devices/uwb/UwbSession.cxx

```diff
@@ -14,12 +16,152 @@ UwbSession::UwbSession(std::weak_ptr<uwb::UwbSessionEventCallbacks> callbacks, w
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_ROUND_USAGE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MULTI_NODE_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_CHANNEL_NUMBER, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_NUMBER_OF_CONTROLEES, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_MAC_ADDRESS, 0 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DST_MAC_ADDRESS, 0 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SLOT_DURATION, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_INTERVAL, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_INDEX, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAC_FCS_TYPE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_ROUND_CONTROL, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_AOA_RESULT_REQ, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_PROXIMITY_NEAR, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_PROXIMITY_FAR, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_ROLE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RFRAME_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PREAMBLE_CODE_INDEX, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SFD_ID, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PSDU_DATA_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PREAMBLE_DURATION, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_TIME_STRUCT, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SLOTS_PER_RR, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_TX_ADAPTIVE_PAYLOAD_POWER, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RESPONDER_SLOT_INDEX, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PRF_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SCHEDULED_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_KEY_ROTATION, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_KEY_ROTATION_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SESSION_PRIORITY, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAC_ADDRESS_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_VENDOR_ID, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STATIC_STS_IV, 6 },
+    { UWB_APP_CONFIG_PARAM_TYPE_NUMBER_OF_STS_SEGMENTS, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAX_RR_RETRY, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_UWB_INITIATION_TIME, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_HOPPING_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_BLOCK_STRIDE_LENGTH, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RESULT_REPORT_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_IN_BAND_TERMINATION_ATTEMPT_COUNT, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SUB_SESSION_ID, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_BPRF_PHR_DATA_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAX_NUMBER_OF_MEASUREMENTS, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_LENGTH, 1 }
+};
+
+std::unordered_map<
+    UWB_APP_CONFIG_PARAM_TYPE,
+    std::tuple<uwb::protocol::fira::UwbConfiguration::ParameterTag,
+        std::function<uint8_t *(std::any, std::size_t, uint8_t *)>>>
+    ddi_service_map{
+        { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_ROLE,
+            { uwb::protocol::fira::UwbConfiguration::ParameterTag::DeviceRole,
+                [](std::any arg, std::size_t size, uint8_t *dst) {
```

Comment:
Why do we have to sacrifice type-safety by using `std::any`? Can't we use `std::visit` to get the appropriate types in the `UwbSessionData` value-map?

Reply:
The use of std::any was before the std::variant pr was merged into develop. I will switch to using variants